cmd/validate.go

```diff
@@ -0,0 +1,78 @@
+package main
+
+import (
+	"errors"
+	"fmt"
+	"github.com/jedib0t/go-pretty/v6/table"
+	"github.com/ldebruijn/graphql-protect/internal/app/config"
+	"github.com/ldebruijn/graphql-protect/internal/business/persisted_operations"
+	"github.com/ldebruijn/graphql-protect/internal/business/protect"
+	"github.com/ldebruijn/graphql-protect/internal/business/schema"
+	"github.com/vektah/gqlparser/v2/gqlerror"
+	"log/slog"
+	"os"
+	"strings"
+)
+
+var ErrValidationErrorsFound = errors.New("errors found during validation")
+
+func validate(log *slog.Logger, cfg *config.Config, shutdown chan os.Signal) error {
+	if !cfg.PersistedOperations.Enabled {
+		err := fmt.Errorf("persisted operations must be enabled to run validate")
+		log.Error("Error running validations", "err", err)
+		return err
+	}
+
+	remoteLoader, err := persisted_operations.RemoteLoaderFromConfig(cfg.PersistedOperations, log)
+	if err != nil && !errors.Is(err, persisted_operations.ErrNoRemoteLoaderSpecified) {
+		log.Warn("Error initializing remote loader", "err", err)
+	}
+
+	po, err := persisted_operations.NewPersistedOperations(log, cfg.PersistedOperations, persisted_operations.NewLocalDirLoader(cfg.PersistedOperations), remoteLoader)
+	if err != nil {
+		log.Error("Error initializing Persisted Operations", "err", err)
+		return nil
+	}
+
+	schemaProvider, err := schema.NewSchema(cfg.Schema, log)
+	if err != nil {
+		log.Error("Error initializing schema", "err", err)
+		return nil
+	}
+
+	protectHandler, err := protect.NewGraphQLProtect(log, cfg, po, schemaProvider, nil)
```

Comment:
protectHandler makes me think this is a HTTP Handler, which it probably also implements, but this is not a context of a HTTP server so it makes me a bit confused?

Reply:
renamed the var